doc/aaa/SONiC RBAC.md

```diff
@@ -0,0 +1,486 @@
+    * *role* : The role associated with the user that is being authorized. This is a string.
+    * *feature* : This is feature that the role is being authorized to access. The granularity of the feature can be :
+        * *feature* - A logical grouping of multiple commands. If the user is authorized to access a particular feature, the column contains the tag associated with that feature.
+        * *entire-system* - If the user is being granted access to the entire system, the column contains *all*
+    * *permissions* : Defines the permissions associated with the role. This is a string.
+        * *none* - This is the default permissions that a role is created with. A role associated with *none* permission cannot access any resources on the system to read, or to modify them.
+        * *read-only* - The role only has read access to the resources associated with the *feature*.
+        * *read-write* - The role has permissions to read and write (create, modify and delete) the resources associated with the *feature*.
+  The PrivilegeTable is keyed on <***role, feature***>
+
+* **ResourceTable**
+  Though the resources are statically tagged with the features that they belong to, a ResourceTable is still needed so as to allow for future extensibility. It is possible that in the future, a customer wants a more granular control over the authorization and wants to either sub partition the features or override the default tagging associated with a feature. The ResourceTable will allow for this support in the future. In the Phase 2, this table will be create using the default tagging associated with the resources.
+    * *resource* : The xpath associated with the resource being accessed. This is a string.
+    * *feature-name* : The tag of the feature this resource belongs to.
+  The ResourceTable is keyed on <***resource, feature***>
+
+* **TenantTable**
+  (To be implemented in Phase 3 or when Multi-tenancy is introduced)
+  In most systems today, a single SONiC system will serve multiple tenants. A tenant is a group of users who have a different privileges for resource instances. As SONiC becomes multitenant, RBAC needs to account for this when authorizing users. The TenantTable is needed to enable this and has the following columns :
+    * *resource* : The xpath associated with the resource being accessed. This is a string.
+    * *tenant* : The tenant for which the resource partitioning is being done. This is a string.
+    * *instances* : The instances of the *resource* allocated to this *tenant*. This is a list of instances.
+  The TenantTable is keyed on <***resource, tenant***>
+
+### 3.2.2 APP DB
+N/A
+
+### 3.2.3 STATE DB
+N/A
+
+### 3.2.4 ASIC DB
+N/A
+
+### 3.2.5 COUNTER DB
+N/A
+
+## 3.3 Switch State Service Design
+### 3.3.1 Orchestration Agent
+N/A
+
+### 3.3.2 Other Process
+N/A
+
+## 3.4 SyncD
+
+N/A. HAM is going to be the single source of truth, and it will take care of
+synchronizing the User Table and /etc/passwd
```

Comment:
So, for remote users, will there be a way to add a special tag to the user entry in the User Table to indicate that it is a remote user that should not be synchronized with /etc/passwd ?

Reply:
It will still synchronize with /etc/passwd, since HAM needs to reserve a UID/GID pair for each user. The password itself won't be saved if it is a remote user - it will be saved as `*`, to indicate that there is no password.